voxgraph/include/voxgraph/tools/data_servers/submap_server.h

```diff
@@ -2,24 +2,25 @@
 #define VOXGRAPH_TOOLS_DATA_SERVERS_SUBMAP_SERVER_H_
 
 #include <std_msgs/Header.h>
-#include <voxgraph_msgs/MapHeader.h>
+#include <cblox_msgs/MapHeader.h>
 
 #include "voxgraph/common.h"
 #include "voxgraph/frontend/submap_collection/voxgraph_submap.h"
 #include "voxgraph/frontend/submap_collection/voxgraph_submap_collection.h"
+#include "voxgraph/frontend/frame_names.h"
 
 namespace voxgraph {
 class SubmapServer {
  public:
   explicit SubmapServer(ros::NodeHandle nh_private);
 
   // Publish maps using the publishers that are members of this server instance
-  void publishSubmap(const VoxgraphSubmap& submap, const ros::Time& timestamp);
-  void publishSubmapTsdf(const VoxgraphSubmap& submap,
+  void publishSubmap(const VoxgraphSubmap::Ptr& submap_ptr, const ros::Time& timestamp);
+  void publishSubmapTsdf(const VoxgraphSubmap::Ptr& submap_ptr,
                          const ros::Time& timestamp);
-  void publishSubmapTsdfAndEsdf(const VoxgraphSubmap& submap,
+  void publishSubmapTsdfAndEsdf(const VoxgraphSubmap::Ptr& submap_ptr,
                                 const ros::Time& timestamp);
-  void publishSubmapSurfacePointcloud(const VoxgraphSubmap& submap,
+  void publishSubmapSurfacePointcloud(const VoxgraphSubmap::Ptr& submap_ptr,
                                       const ros::Time& timestamp);
```

Comment:
Why change to the pointers here?

Reply:
The underlying conversion functions from cblox take the pointer as an argument (see [here](https://github.com/ethz-asl/cblox/blob/master/cblox_ros/include/cblox_ros/submap_conversions.h#L18)), so it seemed like the easier option to me. I could also generate a pointer in the function or change the cblox functions to take a const ref, as you prefer.

Reply:
If possible could we change the underlying functions in cblox to take const refs? The problem, in my view, with taking shared_ptrs in contexts like this is that it forces the caller to have a shared pointer as opposed to a reference, and this requirement propagates all the way up the call tree. 

Just to make sure we're not totally off here, I checked voxblox, and [they also take const refs](https://github.com/ethz-asl/voxblox/blob/master/voxblox_ros/include/voxblox_ros/conversions.h#L148).

Reply:
Everything's back to const refs and the cblox update is in this [PR](https://github.com/ethz-asl/cblox/pull/35).